### PR TITLE
Message Archive scenario: Bot with RSC permission - C#

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**
Bot with RSC permission can fetch all the messages from a group chat, archive it to text file and send it to user in personal scope.

### CODE REVIEW

File: samples/bot-groupchat-messages-withRSC/csharp/FetchGroupChatMessagesWithRSC/helper/GetChatHelper.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                throw ex;
+            }
+        }
+
+        public static async Task SendGroupChatMessage(ITurnContext turnContext, long fileSize,string microsoftAppId, string microsoftAppPassword,  CancellationToken cancellationToken)
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/csharp/FetchGroupChatMessagesWithRSC/helper/GetChatHelper.cs:44]: One line space before it.

Also add summary for all the methods, if required.

File: samples/bot-groupchat-messages-withRSC/csharp/FetchGroupChatMessagesWithRSC/wwwroot/chat.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from: MOD Administrator
+text: <div><div><at id="0">RSC Sample Groupchat</at> getchat</div>
+</div>
+at: 9/23/2021 4:20:56 PM +00:00
+from: MOD Administrator
+text: <div><div><at id="0">RSC Sample Groupchat</at> getchat</div>
+</div>
+at: 9/23/2021 4:13:15 PM +00:00
+from: bot
+text: <p>hello</p>
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/csharp/FetchGroupChatMessagesWithRSC/wwwroot/chat.txt:35]: Is it required to push this data as well or empty file??

File: samples/bot-groupchat-messages-withRSC/csharp/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+﻿﻿﻿Using this C# sample, a bot can receive all channel messages with RSC without @mention.
```
Comment (v-chetsh) [inline samples/bot-groupchat-messages-withRSC/csharp/README.md:1]: Please make sure that this readme file is linked with the root readme file.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Added my feedback, please look into it.

File: README.md [hunk 4]
```diff
@@ -218,6 +218,7 @@ Here are all the samples which are deployed on Microsoft Azure and you can try i
 [messageswithrsc#csharp]:samples/bot-receive-channel-messages-withRSC/csharp
 [messageswithrsc#js]:samples/bot-receive-channel-messages-withRSC/nodejs
 [app-in-meeting#cs]:samples/app-in-meeting/csharp
+[fetchgroupchatmessages#nodejs]:samples/bot-groupchat-messages/nodejs
```
Comment (Wajeed-msft) [inline README.md:221]: Could you please rename this to bot-archive-groupchat-messages?

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**